Marlin/src/HAL/STM32/u8g/LCD_defines.h

```diff
@@ -25,8 +25,13 @@
  * STM32 LCD-specific defines
  */
 
-uint8_t u8g_com_std_sw_spi_fn(u8g_t *u8g, uint8_t msg, uint8_t arg_val, void *arg_ptr);         // See U8glib-HAL
-uint8_t u8g_com_stm32duino_hw_spi_fn(u8g_t *u8g, uint8_t msg, uint8_t arg_val, void *arg_ptr);  // See U8glib-HAL
+#if ALL(HAS_MARLINUI_U8GLIB, FORCE_SOFT_SPI)
```

Comment:
I don't think this is ever true due to file include order.

One of the problematic hardware configurations I have (and was hoping this PR would help) is on a BIQU Hurakan with a `BTT_MINI_12864` & , `BOARD_BTT_MANTA_M4P_V2_1` (STM32G0). Both `HAS_MARLINUI_U8GLIB` & `FORCE_SOFT_SPI` are set elsewhere, but they don't appear to be true at the same time ahead of this (`LCD_defines.h`) file.

I threw in some warnings as a check:

<img width="969" alt="image" src="https://github.com/MarlinFirmware/Marlin/assets/13375512/aac95f91-1294-43bf-93e5-c8fc896310a6">

In either case, it doesn't fix the SPI issues I'm having with this LCD/hardware combination, but I thought I'd mention it.